chain/beacon/sync_manager.go

```diff
@@ -91,29 +104,31 @@ func (s *SyncManager) Stop() {
 
 type requestInfo struct {
 	nodes []net.Peer
+	from  uint64
 	upTo  uint64
 }
 
 // RequestSync asks the sync manager to sync up with those peers up to the given
 // round. Depending on the current state of the syncing process, there might not
 // be a new process starting (for example if we already have the round
 // requested). upTo == 0 means the syncing process goes on forever.
-func (s *SyncManager) RequestSync(nodes []net.Peer, upTo uint64) {
+func (s *SyncManager) RequestSync(upTo uint64, nodes []net.Peer) {
 	s.newReq <- requestInfo{
 		nodes: nodes,
 		upTo:  upTo,
 	}
 }
 
-func (s *SyncManager) Run() {
+// Run handles non-blocking sync requests coming from the regular operation of the daemon
+func (s *SyncManager) Run(ctx context.Context) {
```

Comment:
Now that we are adding this one, we should listen for `ctx.Done()` as well and stop sync manager in case it is done.

Reply:
I'm also wondering what' sthe usage of the stop channel now - maybe it doesn't need to be there anymore, otherwise that means we have two mechanisms to stop the sync manager operations.

Reply:
That's even more confusing now - What you expect by giving a context in a method is that cancelling this context will quit the methods. Now you tell me it's not the case. If the context only removes the _current_ operation, can't we put it inside the sync request struct that gets sent into the chan ? For example. Maybe other ways, but Im kinda reluctant of changing the expectation of the APi. 
* If there is a stop() method it stops all
* If a method expects a context, cancelling that context will cancel what the method is doing - all of it -

Reply:
I'll remove it since it isn't really needed anymore since I've split the forced resync out of the `Run` loop anyway.